### ISSUE TITLE
Provide support for client endpoint authorization

### ISSUE THREAD

**Issue by estein9825** [terraform-providers/terraform-provider-aws#7523]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

Would like the ability to create an endpoint authorization rule for a network or internet destination

### New or Affected Resource(s)

* aws_ec2_client_vpn_authorization

### Potential Terraform Configuration

```hcl
# associate the appropriate subnet to the endpoint
resource "aws_ec2_client_vpn_authorization" "client_vpn_authorization_internet" {
  description = "Internet access"
  client_vpn_endpoint_id = "${aws_ec2_client_vpn_endpoint.client_endpoint.id}"
  cidr_block              = "0.0.0.0/0"
  authorize_all_groups = true
}

# associate the appropriate subnet to the endpoint
resource "aws_ec2_client_vpn_authorization" "client_vpn_authorization_network" {
  description = "Internet access"
  client_vpn_endpoint_id = "${aws_ec2_client_vpn_endpoint.client_endpoint.id}"
  cidr_block              = "10.4.0.0/16"
  access_group_id = "S-1-5-32-544"
}
```
**Note**: access_group_id is the Active Directory SID available via the command `Get-ADGroup -Filter 'Name -eq "<Name of the AD Group>"'`

### References

[CLI Documentation](https://docs.aws.amazon.com/cli/latest/reference/ec2/authorize-client-vpn-ingress.html)
[AWS Documentation](https://docs.aws.amazon.com/vpn/latest/clientvpn-admin/cvpn-working-rules.html)

**Comment by yardensachs** [comment]
Is this a duplicate of #7494 ?

**Comment by Veramkovich** [comment]
Any update on this?

**Comment by yardensachs** [comment]
This can be closed. PR #13950 had merged in this request

**Comment by bflad** [comment]
Closing as noted above the `aws_ec2_client_vpn_authorization_rule` resource was released in version 2.70.0 of the Terraform AWS Provider.
